samples/_opengl/NvidiaMulticast/src/NvidiaMulticastApp.cpp

```diff
@@ -0,0 +1,270 @@
+}
+
+void NvidiaMulticastApp::draw()
+{
+	gl::clear();
+	gl::setMatrices( mCam );
+
+	if( settings["copy_to_primary"].asBool() ) {
+		{
+			gl::ScopedFramebuffer pusFbo{ mFbo };
+			gl::clear();
+#if defined( ASYMMETRICAL_CAMERAS )
+			gl::viewport( ivec2( getWindowWidth() / 2, getWindowHeight() ) );
+#else
+			gl::viewport( app::getWindowSize() );
+			multicast::viewport( mGPUs[0], ivec2( 0 ), getWindowSize() );
+			multicast::viewport( mGPUs[1], ivec2( -getWindowWidth() / 2, 0 ), getWindowSize() );
+#endif
+			drawScene();
+		}
+
+		multicast::copyImageSubData( mFbo->getColorTexture(), mTextureSecondary, mGPUs[1], mGPUs[0] );
+
+		multicast::signalWaitSync( mGPUs[1], mGPUs[0] );
+		multicast::enableRenderMask( mGPUs[0] );
+
+		gl::viewport( app::getWindowSize() );
+		gl::setMatricesWindow( app::getWindowSize() );
+		gl::draw( mFbo->getColorTexture() );
+		gl::translate( vec2( mFbo->getWidth(), 0.0f ) );
+		gl::draw( mTextureSecondary );
+
+		//GPU 1 should wait for GPU0 to be done with screen rendering.
+		multicast::signalWaitSync( mGPUs[0], mGPUs[1] );
+		multicast::disableRenderMask();
+	}
+	else {
+#if defined( ASYMMETRICAL_CAMERAS )
+		multicast::viewport( mGPUs[0], ivec2( 0 ), ivec2( getWindowWidth() / 2, getWindowHeight() ) );
+		multicast::viewport( mGPUs[1], ivec2( getWindowWidth() / 2, 0 ), ivec2( getWindowWidth() / 2, getWindowHeight() ) );
+#else
+		gl::viewport( app::getWindowSize() );
+		multicast::scissor( mGPUs[0], ivec2( 0 ), ivec2( getWindowWidth() / 2, getWindowHeight() ) );
+		multicast::scissor( mGPUs[1], ivec2( getWindowWidth() / 2, 0 ), ivec2( getWindowWidth() / 2, getWindowHeight() ) );
+#endif
+		drawScene();
+	}
+}
+
+void NvidiaMulticastApp::drawScene()
+{
+#if defined( ASYMMETRICAL_CAMERAS )
+	for( size_t i = 0; i < mGPUs.size(); ++i ) {
+		auto cameraTile = mCam.subdivide( uvec2( 1, mGPUs.size() ), uvec2( 0, i ) );
+		mCameraMatrices.view = cameraTile.getViewMatrix();
+		mCameraMatrices.projection = cameraTile.getProjectionMatrix();
+		multicast::bufferSubData( mCamUbo, sizeof( CameraMatrices ), &mCameraMatrices, mGPUs.at(i) );
+	}
+#endif
+	gl::ScopedTextureBind push( mTexture, 0 );
+	gl::ScopedSsboBind pushBuffer{ mSsbo, 0 };
+	mBatch->drawInstanced( (GLsizei)mInstanceCount );
+}
+
+void NvidiaMulticastApp::keyDown( KeyEvent event )
+{
+	if( event.isControlDown() && event.getCode() == KeyEvent::KEY_q ) {
+		quit();
+	}
+}
+
+RendererGl::Options getRendererGlOptions()
+{
+	// Load settings first!
+	std::ifstream in{ getAssetPath( "settings.json" ) };
+	Json::Reader reader;
+	reader.parse( in, settings );
+
+	RendererGl::Options options;
+	if( settings["multicast_multidisplay"].asBool() ) {
+		//! This options should be use on Quadro setups with Mosaic enabled, with each display connected to independent GPUs.
+		options.multiGpuMultiDisplayMulticastNV();
+	}
+	else {
+		//! This option should be used on GeForce setups, with SLI "Maximize 3D Performance" enabled.
```

Comment:
This path can also be used on Quadro, same setting in the Control Panel

Reply:
Thanks for reviewing! Both notes should be addressed in the latest commit.